docs/BRANCHING.md

```diff
@@ -0,0 +1,51 @@
+# Branching Strategy for releases
+
+## General Considerations
+
+- Generally the master branch is the one where current development is happening.
+- One objective would be to arrive to a point where the master branch can be released very often (every day?).
+  This means the master branch is releasable at all times. I believe this strategy is called **Continuous Delivery**.
+- At this point we don't think it's possible to do that until the testing infrastructure has been improved.
+- Nevertheless, the branching strategy takes this objective into account.
+- To further support this objective, the strategy wil try to minimize the number of long-lasting branches.
+
+## Release numbering
+- example 4.3.0
+- The first number (4 in this example) is incremented for major changes in the software. What major change mean has to be decided.
+- The second number (3 in this example) is incremented for minor changes in the software.
+- The third number (0 in this example) is used for maintenance releases.
+  A maintenance is created if some bug has to be corrected before the next minor release.
+![](BranchingDiagram.png)
+## Feature/fix branches
```

Comment:
@jcpitre I think I'm fuzzy on when we'd use a feature branch in the validator. Wouldn't we generally rely on release branches?

Reply:
Feature branches are the way we already do development. Create a short lived branch corresponding to an issue, then merge it back to master. So it's not really related to releases. Should I remove this part?

Reply:
Nope! Just context I didn't have - makes sense now :)